kratos/expression/variable_expression_data_io.cpp

```diff
@@ -196,6 +409,23 @@ void VariableExpressionDataIO<Vector>::Read(
     }
 }
 
+template<>
+void VariableExpressionDataIO<std::vector<Vector>>::Read(
+    RawLiteralFlatExpression& rExpression,
+    const IndexType EntityIndex,
+    const std::vector<Vector>& Value) const
+{
+    const auto& r_shape = rExpression.GetItemShape();
+
+    const IndexType entity_data_begin_index = EntityIndex * r_shape[0] * r_shape[1];
```

Comment:
make sure the `r_shape` is of size 2

Reply:
as mentioned in the previous comment, added a full debug error and a comment.